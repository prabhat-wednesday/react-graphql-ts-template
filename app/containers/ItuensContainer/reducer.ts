import { createSlice } from '@reduxjs/toolkit';

export interface ApiResponseState {
  loading: boolean;
  dataToShow: [];
  error: string;
  searchTerm: string;
}

export const initialState: ApiResponseState = {
  loading: false,
  dataToShow: [],
  error: '',
  searchTerm: ''
};

const ituneSlice = createSlice({
  name: 'Itune',
  initialState,
  reducers: {
    getSearchTerm: (state, action) => {
      state.searchTerm = action.payload;
    },
    getDataToShow: (state, action) => {
      state.dataToShow = action.payload;
    }
  }
});

export const { getSearchTerm, getDataToShow } = ituneSlice.actions;

export default ituneSlice.reducer;
