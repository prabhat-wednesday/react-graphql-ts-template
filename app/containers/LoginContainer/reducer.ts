import { createSlice } from '@reduxjs/toolkit';

export interface loginProviderState {
  userData: {};
  isLogin: boolean;
}

export const initialState: loginProviderState = {
  userData: {},
  isLogin: false
};

const LoginProviderSlice = createSlice({
  name: 'LoginProviderSlice',
  initialState,
  reducers: {
    requestGetUserData: (state, action) => {
      state.userData = action.payload;
      state.isLogin = true;
    },
    logoutUser: (state) => {
      state.userData = {};
      state.isLogin = false;
    }
  }
});

export const { requestGetUserData, logoutUser } = LoginProviderSlice.actions;

export default LoginProviderSlice.reducer;
