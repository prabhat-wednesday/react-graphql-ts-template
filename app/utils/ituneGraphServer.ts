import ApolloClient, { DocumentNode, InMemoryCache } from 'apollo-boost';

export const client = new ApolloClient({
  uri: 'https://itunes.apple.com/search?term=',
  cache: new InMemoryCache()
});

export interface GqlQueryReponse<Data> {
  data?: Data;
  error?: any;
  ok: boolean;
}

export const getQueryResponse = <Data, Variables>(
  query: DocumentNode,
  variables?: Variables
): Promise<GqlQueryReponse<Data>> => {
  console.log(query, 'in graphql server');
  return client
    .query<Data, Variables>({ query, variables })
    .then((res) => {
      console.log(res, 'in graphql');
      if (res.errors) {
        console.log(res);
        return { error: res.errors, ok: false };
      } else {
        console.log(res);
        return { data: res.data, ok: true };
      }
    })
    .catch((err) => {
      return { error: err, ok: false };
    });
};
