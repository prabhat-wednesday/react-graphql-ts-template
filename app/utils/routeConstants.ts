export type RouteConstant = {
  route: string;
  exact?: boolean;
  isProtected?: boolean;
  props?: object;
};

const routeConstants: Record<string, RouteConstant> = {
  itune: {
    route: '/',
    exact: true
  },
  song: {
    route: '/song/:trackId',
    exact: true
  }
};

export type RouteKeys = keyof typeof routeConstants;

export default routeConstants;
