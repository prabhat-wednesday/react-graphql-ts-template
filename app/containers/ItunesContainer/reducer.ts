import { prepare } from '@app/utils';
import { createSlice } from '@reduxjs/toolkit';
import { song } from './types';

export interface ituneResponseState {
  loading: boolean;
  songData: {
    resultCount: number;
    results: song[];
  };
  songListError: string;
}

export const initialState: ituneResponseState = {
  loading: false,
  songData: { resultCount: 0, results: [] },
  songListError: ''
};

const ituneSlice = createSlice({
  name: 'Itune',
  initialState,
  reducers: {
    requestGetSongList: {
      reducer: (state) => {
        state.loading = true;
      },
      prepare
    },
    successGetSongList: (state, action) => {
      state.songData = action.payload;
      state.songListError = '';
      state.loading = false;
    },
    failureGetSongList: (state, action) => {
      state.songData.resultCount = 0;
      state.songData.results = [];
      state.loading = false;
      state.songListError = action.payload;
    }
  }
});

export const { requestGetSongList, successGetSongList, failureGetSongList } = ituneSlice.actions;

export default ituneSlice.reducer;
