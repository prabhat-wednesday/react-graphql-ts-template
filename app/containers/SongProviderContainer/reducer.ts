import { prepare } from '@app/utils';
import { createSlice } from '@reduxjs/toolkit';
import { Song } from '../ItunesContainer/types';

interface songProviderState {
  songData: {
    results: Song[];
  };
  loading: boolean;
  songListError: string;
}

export const initialState: songProviderState = {
  songData: {
    results: []
  },
  loading: false,
  songListError: ''
};

const SongProviderSlice = createSlice({
  name: 'SongProviderSlice',
  initialState,
  reducers: {
    requestGetSongList: {
      reducer: (state) => {
        state.loading = true;
      },
      prepare
    },
    successGetSongList: (state, action) => {
      state.songData.results = action.payload.results;
      state.songListError = '';
      state.loading = false;
    },
    failureGetSongList: (state, action) => {
      state.songData.results = [];
      state.loading = false;
      state.songListError = action.payload;
    }
  }
});

export const { requestGetSongList, successGetSongList, failureGetSongList } = SongProviderSlice.actions;

export default SongProviderSlice.reducer;
