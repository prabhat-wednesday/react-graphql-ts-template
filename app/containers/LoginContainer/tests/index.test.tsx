import React from 'react';
import { fireEvent } from '@testing-library/react';
import { renderProvider, timeout } from '@app/utils/testUtils';
import { requestGetUserData } from '../reducer';
import { LoginContainerTest as LoginContainer, LoginContainerProps, mapDispatchToProps } from '..';

describe('<LoginContainer tests', () => {
  let submitSpy: jest.Mock;
  let defaultProps: LoginContainerProps;
  beforeEach(() => {
    defaultProps = {
      dispatchUserData: submitSpy
    };
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<LoginContainer {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should validate the mapDispatchToProps action', () => {
    const dispatchUserDataSpy = jest.fn();
    const payload = {
      username: 'Prabhat Singh',
      emailId: 'singhprabhat',
      password: 'qwerty1234'
    };
    const action = {
      dispatchUserData: requestGetUserData(payload)
    };
    const props = mapDispatchToProps(dispatchUserDataSpy);
    props.dispatchUserData(payload);
    expect(dispatchUserDataSpy).toHaveBeenCalledWith(action.dispatchUserData);
  });

  it('should render on the screen', () => {
    const { getByTestId } = renderProvider(<LoginContainer {...defaultProps} />);
    const LoginForm = getByTestId('loginForm');
    expect(LoginForm).toBeInTheDocument();
  });

  it('should check if form get submit after clicking on submit button', async () => {
    const submitSpy = jest.fn();
    const { getByTestId, getAllByTestId } = renderProvider(<LoginContainer dispatchUserData={submitSpy} />);
    const onFinishSpy = jest.fn();
    const LoginForm = getByTestId('loginForm');
    const LoginInputs = getAllByTestId('formInput') as HTMLInputElement[];
    LoginForm.onsubmit = onFinishSpy;

    fireEvent.change(LoginInputs[0], {
      target: { value: 'Prabhat Singh' }
    });
    fireEvent.change(LoginInputs[1], {
      target: { value: 'singhprabhat@gmail.com' }
    });
    fireEvent.change(LoginInputs[2], {
      target: { value: 'qwerty1234' }
    });
    fireEvent.submit(LoginForm);
    await timeout(500);
    expect(onFinishSpy).toBeCalled();
    expect(submitSpy).toBeCalled();
  });
});
