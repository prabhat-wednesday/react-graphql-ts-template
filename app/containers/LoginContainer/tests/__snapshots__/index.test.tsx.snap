// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="LoginContainer__Container-sk1ioa-0 jvcMSm"
    >
      <div
        class="ant-steps ant-steps-horizontal StepsComponent__CustomStep-ymgfaa-0 kQDDmx ant-steps-label-horizontal"
      >
        <div
          class="ant-steps-item ant-steps-item-process ant-steps-item-active"
        >
          <div
            class="ant-steps-item-container"
          >
            <div
              class="ant-steps-item-tail"
            />
            <div
              class="ant-steps-item-icon"
            >
              <span
                class="ant-steps-icon"
              >
                1
              </span>
            </div>
            <div
              class="ant-steps-item-content"
            >
              <div
                class="ant-steps-item-title"
              >
                Username
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-steps-item ant-steps-item-wait"
        >
          <div
            class="ant-steps-item-container"
          >
            <div
              class="ant-steps-item-tail"
            />
            <div
              class="ant-steps-item-icon"
            >
              <span
                class="ant-steps-icon"
              >
                2
              </span>
            </div>
            <div
              class="ant-steps-item-content"
            >
              <div
                class="ant-steps-item-title"
              >
                Emailid
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-steps-item ant-steps-item-wait"
        >
          <div
            class="ant-steps-item-container"
          >
            <div
              class="ant-steps-item-tail"
            />
            <div
              class="ant-steps-item-icon"
            >
              <span
                class="ant-steps-icon"
              >
                3
              </span>
            </div>
            <div
              class="ant-steps-item-content"
            >
              <div
                class="ant-steps-item-title"
              >
                Password
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="DynamicForm__FormContainer-sc-1ttvbzz-0 VZLZL"
      >
        <form
          autocomplete="off"
          class="ant-form ant-form-vertical"
          data-testid="login-form"
          id="basic"
        >
          <div>
            <div
              class="ant-row ant-form-item FormInput__CustomFormItem-ofxmhy-1 cElTSv"
            >
              <div
                class="ant-col ant-form-item-label"
              >
                <label
                  class="ant-form-item-required"
                  for="basic_username"
                  title="Username"
                >
                  Username
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input FormInput__StyledInput-ofxmhy-0 iWohJH"
                      data-testid="formInput"
                      id="basic_username"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        style="display: flex;"
      >
        <button
          class="ant-btn ant-btn-primary Button__B-sc-10epy5l-0 jyXgnZ"
          messageid="next"
          type="button"
        >
          <p
            class="T__StyledText-gcrjip-0 gIxZML"
            data-testid="t"
          >
            Next
          </p>
        </button>
      </div>
    </div>
  </div>
</body>
`;
